tests/unit/acl-v2.tcl

```diff
@@ -116,11 +116,32 @@ start_server {tags {"acl external:skip"}} {
         assert_match "*NOPERM*key*" $err
     }
 
-    test {Validate read and write permissions format} {
+    test {Validate read and write permissions format - empty permission} {
         catch {r ACL SETUSER key-permission-RW %~} err
         set err
     } {ERR Error in ACL SETUSER modifier '%~': Syntax error}
 
+    test {Validate read and write permissions format - empty selector} {
+        catch {r ACL SETUSER key-permission-RW %} err
+        set err
+    } {ERR Error in ACL SETUSER modifier '%': Syntax error}
+
+    test {Validate read and write permissions format - empty pattern} {
+        # Empty pattern results with R/W access to no key
+        r ACL SETUSER key-permission-RW on nopass %RW~ +@all
+        $r2 auth key-permission-RW password
+        catch {$r2 SET x 5} err
+        set err
+    } {NOPERM No permissions to access a key}
+
+    test {Validate read and write permissions format - no pattern} {
+        # No pattern results with R/W access to no key (currently we accept this syntax error)
+        r ACL SETUSER key-permission-RW on nopass %RW +@all
+        $r2 auth key-permission-RW password
+        catch {$r2 SET x 5} err
+        set err
+    } {NOPERM No permissions to access a key}
```

Comment:
Did a quick check locally both the ACL rules get simplified to the following i.e. `~` no key. I also think it's too late to error out on the no pattern with a selector. Possibly add more assertion to the test case to verify the simplified ACL rule on the server. (can be done later).
```
user key-permission-RW on nopass sanitize-payload ~ resetchannels +@all
```

Reply:
I'll make a note. I also think the comment is wrong, but was just pulling it verbatim to retain origin from the Redis BSD Stream.

Reply:
https://github.com/valkey-io/valkey/pull/1515